tensorboard/webapp/runs/store/runs_reducers_test.ts

```diff
@@ -965,5 +966,67 @@ describe('runs_reducers', () => {
         ])
       );
     });
+
+    it('sets initial groupby to EXPERIMENT on experiment comparion', () => {
+      const state = buildRunsState({
+        initialGroupBy: {key: GroupByKey.RUN},
+      });
+
+      const nextState = runsReducers.reducers(
+        state,
+        navigated({
+          before: buildRoute({
+            routeKind: RouteKind.EXPERIMENT,
+          }),
+          after: buildCompareRoute(['eid1:run1', 'eid1:run2']),
+        })
+      );
+
+      expect(nextState.data.initialGroupBy.key).toEqual(GroupByKey.EXPERIMENT);
+    });
+
+    it('preserves userSetGroupByKey on comparision view navigation', () => {
+      const state = buildRunsState({
+        initialGroupBy: {key: GroupByKey.EXPERIMENT},
+        userSetGroupByKey: GroupByKey.RUN,
+      });
+
+      const state2 = runsReducers.reducers(
+        state,
+        navigated({
+          before: buildCompareRoute(['eid1:run1', 'eid2:run2']),
+          after: buildCompareRoute(['eid3:run3', 'eid4:run4']),
+        })
+      );
+
+      const state3 = runsReducers.reducers(
+        state2,
+        navigated({
+          before: buildCompareRoute(['eid3:run3', 'eid4:run4']),
+          after: buildCompareRoute(['eid1:run1', 'eid2:run2']),
+        })
+      );
+
+      expect(state3.data.initialGroupBy.key).toEqual(GroupByKey.EXPERIMENT);
+      expect(state3.data.userSetGroupByKey).toEqual(GroupByKey.RUN);
```

Comment:
Consider wrapping all new tests around a contextual `describe` block and say something like "when freshly navigating"